src/client/opamClient.ml

```diff
@@ -635,6 +635,57 @@ let init_checks ?(hard_fail_exn=true) init_config =
   if hard_fail && hard_fail_exn then OpamStd.Sys.exit_because `Configuration_error
   else not (soft_fail || hard_fail)
 
+let git_for_windows_check =
+  if Sys.win32 then
```

Comment:
Shouldn’t this code be also ran on Cygwin?
```suggestion
  if Sys.win32 || Sys.cygwin then
```

Reply:
I personally agree with @kit-ty-kate but I'm not a Cygwin user. I could see it going either way; Cygwin (not MSYS2) is meant as a full Unix replacement including userland tools.

Reply:
No - for the same reason as we're advising users aiming for native Windows opam experience to use Git for Windows, users of Cygwin's opam should be using Cygwin's git (or at least, we shouldn't be advising the contrary). Cygwin's opam package is maintained, as well.